### PR TITLE
WIP: Fix libpsl-native to build and pass tests on FreeBSD

### PR DESCRIPTION
**Pull request by mateusrodrigues**
## PR Summary

As cited in #7207, these are the changes I've made to libpsl-native to fix all tests on FreeBSD. The only two files that needed changes were `src/getuserfrompid.cpp` and `src/isfile.cpp`:

- In `isfile.cpp`, I've changed the logic to verify whether something is a file or not by using a macro available in `sys/stat.h`. This changed allowed `IsFileTest.RootIsFile` to pass.
- In `getuserfrompid.cpp`, a FreeBSD-specific logic wasn't defined, so NULL was being returned every time. Therefore, I've added a logic similar to the macOS one with a subtle difference in the return statement. This allowed `GetUserFromPid.Success` to pass.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
Did we migrate the psl to https://github.com/PowerShell/PowerShell-Native?

**Comment by iSazonov** [comment]
After merging #7129 we will no longer use `src/isfile.cpp`.

**Comment by TravisEz13** [comment]
@adityapatwardhan  Can we answer @iSazonov 's question?  When should we stop taking PRs here?

**Comment by adityapatwardhan** [comment]
@mateusrodrigues Could you submit this PR to https://github.com/PowerShell/PowerShell-Native. We are moving the native code to a separate repo. Thanks for your contribution.

File: src/libpsl-native/src/isfile.cpp [hunk 1]
```diff
@@ -31,5 +31,11 @@ bool IsFile(const char* path)
     assert(path);
 
     struct stat buf;
-    return lstat(path, &buf) == 0;
+    if (lstat(path, &buf) == -1)
+    {
+        // TODO: throw error on path doesn't exist?
+        return false;
```
[the remark this answers is not in the document]
Comment (dantraMSFT) [reply src/libpsl-native/src/isfile.cpp]: The unit test test-isfile.cpp currently expects false for a file that does not exit. See https://github.com/PowerShell/PowerShell/blob/master/src/libpsl-native/test/test-isfile.cpp

[the remark this answers is not in the document]
Comment (dantraMSFT) [reply src/libpsl-native/src/isfile.cpp]: Also, I would suggest testing for path before using it; something like the following:

if (path == NULL || lstat(path, &buf) == -1)

**Comment by mateusrodrigues** [comment]
@adityapatwardhan just moved the PR over

**Comment by adityapatwardhan** [comment]
@mateusrodrigues Thanks! I will close this PR.

**Pull request closed without merging**